test/integration/services/dealer/hedge-price.spec.ts

```diff
@@ -0,0 +1,1892 @@
+  const probeResult = await Payments.getLightningFeeEstimation({
+    uncheckedPaymentRequest: lnInvoice.paymentRequest,
+    walletId: newUsdWallet.id,
+  })
+  if (probeResult instanceof Error) throw probeResult
+  const payResult = await Payments.payInvoiceByWalletId({
+    uncheckedPaymentRequest: lnInvoice.paymentRequest,
+    memo: null,
+    senderWalletId: newUsdWallet.id,
+    senderAccount: newAccount,
+  })
+  if (payResult instanceof Error) throw payResult
+  const afterUsd = await getBalanceHelper(newUsdWallet.id)
+  const diff = (beforeUsd - afterUsd) as CurrencyBaseAmount
+  return diff
+}
+
+const getBtcForUsdEquivalentNoAmountInvoice = async ({
+  btcPaymentAmount,
+  accountAndWallets,
+}: {
+  btcPaymentAmount: BtcPaymentAmount
+  accountAndWallets: AccountAndWallets
+}): Promise<CurrencyBaseAmount | ZeroAmountForUsdRecipientError> => {
+  const { newBtcWallet, newUsdWallet, newAccount } = accountAndWallets
+
+  const lnInvoice = await Wallets.addInvoiceNoAmountForSelf({
+    walletId: newUsdWallet.id,
+  })
+  if (lnInvoice instanceof Error) throw lnInvoice
+
+  const beforeBtc = await getBalanceHelper(newBtcWallet.id)
+  const result = await Payments.payNoAmountInvoiceByWalletId({
+    amount: Number(btcPaymentAmount.amount),
+    uncheckedPaymentRequest: lnInvoice.paymentRequest,
+    memo: null,
+    senderWalletId: newBtcWallet.id,
+    senderAccount: newAccount,
+  })
+  if (result instanceof Error) {
+    if (!(result instanceof ZeroAmountForUsdRecipientError)) throw result
+    return result
+  }
+  const afterBtc = await getBalanceHelper(newBtcWallet.id)
+  const diff = (beforeBtc - afterBtc) as CurrencyBaseAmount
+  return diff
+}
+
+const getBtcForUsdEquivalentNoAmountInvoiceAndProbe = async ({
+  btcPaymentAmount,
+  accountAndWallets,
+}: {
+  btcPaymentAmount: BtcPaymentAmount
+  accountAndWallets: AccountAndWallets
+}): Promise<CurrencyBaseAmount | ZeroAmountForUsdRecipientError> => {
+  const { newBtcWallet, newUsdWallet, newAccount } = accountAndWallets
+
+  const lnInvoice = await Wallets.addInvoiceNoAmountForSelf({
+    walletId: newUsdWallet.id,
+  })
+  if (lnInvoice instanceof Error) throw lnInvoice
+
+  const beforeBtc = await getBalanceHelper(newBtcWallet.id)
+
+  const probe = await Payments.getNoAmountLightningFeeEstimation({
+    amount: Number(btcPaymentAmount.amount),
+    uncheckedPaymentRequest: lnInvoice.paymentRequest,
+    walletId: newBtcWallet.id,
+  })
+  if (probe instanceof Error) {
+    if (!(probe instanceof ZeroAmountForUsdRecipientError)) throw probe
+    return probe
+  }
+
+  const result = await Payments.payNoAmountInvoiceByWalletId({
+    amount: Number(btcPaymentAmount.amount),
+    uncheckedPaymentRequest: lnInvoice.paymentRequest,
+    memo: null,
+    senderWalletId: newBtcWallet.id,
+    senderAccount: newAccount,
+  })
+  if (result instanceof Error) {
+    if (!(result instanceof ZeroAmountForUsdRecipientError)) throw result
+    return result
+  }
```

Comment:
apply to all diff methods, if return zero amount error is required then probably is cleaner do something like this:
```
if (result instanceof ZeroAmountForUsdRecipientError) return result
if (result instanceof Error) throw result
```

Reply:
Done: https://github.com/GaloyMoney/galoy/pull/1846/commits/4ebd8ddd7b4fc84653a2c1d96c36044d73429e30